### ISSUE TITLE
Generate code by building an AST using https://github.com/dart-lang/sdk/tree/master/pkg/analyzer

### ISSUE THREAD

**Issue by alexeieleusis** [polux/adts#6]
This is actually a question, I wonder if it would be better in any way to use the analyzer to generate the code.

**Comment by polux** [comment]
Yes I think that would be nice. Nothing like this was available back in the days when I wrote this tool (also to be frank I wrote this in a rather quick and dirty way so even if it had I might not have used it), but now that there is I think we should use it: it would probably help preventing accidental name captures.

**Comment by alexeieleusis** [comment]
There is also https://github.com/dart-lang/code_builder
